### PR TITLE
Fix bugs, enhancement and standardisation

### PR DESCRIPTION
**Pull request by TPH777**
### Bugs fix: 
1. `list tags` will now work with two or more words tag.
2. `|` will no longer be allowed in description for `add` command.

### Enhancement:
1. `list tags` to be case insensitive.
2. Exceptions are more specific for `find` command.
3. Edit @TPH777 PPP

### Standardisation:
1. Test that can have both `income` or `spending` input will have one test for each.
2. Expected output of failed test will be named in terms of the error message, instead of the exception.
3. `WiagiMissingParamsException` is used for all instances of missing parameters instead of `WiagiInvalidInputException`.
4. Replace all occurence of `/t` to `TAB`.
5. Replace all occurence of `System.lineSeparator()` to `NEXT_LINE`.
6. Replace all occurence of `LocalDate.now()` to `TODAY`.

### CODE REVIEW

File: src/main/java/seedu/commands/FindCommand.java [hunk 1]
```diff
@@ -111,20 +116,24 @@ private <T extends EntryType> ArrayList<T> getMatchingAmount(String findValue, A
         double lower;
         double upper;
         if (findValue.contains(FIND_RANGE_DIVIDER)) { // range
-            String[] range = findValue.split(FIND_RANGE_DIVIDER);
-            lower = CommandUtils.formatAmount(range[0], FIND_COMMAND_FORMAT);
-            upper = CommandUtils.formatAmount(range[1], FIND_COMMAND_FORMAT);
+            String[] rangeValues = extractRangeValues(findValue);
+            lower = CommandUtils.formatAmount(rangeValues[0], FIND_COMMAND_FORMAT);
+            upper = CommandUtils.formatAmount(rangeValues[1], FIND_COMMAND_FORMAT);
```
Comment (wongwh2002) [inline src/main/java/seedu/commands/FindCommand.java:121]: should we refractor the 0s and 1s in the code

Comment (TPH777) [reply src/main/java/seedu/commands/FindCommand.java:121]: ok will do in the next commit

File: src/test/java/seedu/commands/FindCommandTest.java [hunk 2]
```diff
@@ -98,6 +101,27 @@ public void execute_invalidDateFormat_expectIllegalInputExceptionThrown() {
...
+
+    @Test
+    public void execute_longFromValue_expectIllegalInputExceptionThrown() {
+        commandInputForTest("find spending date 2024/10/10 extraString to 2024-12-12", incomes, spendings);
+        assertEquals(TAB + LONG_FIND_FROM_VALUE + FIND_COMMAND_FORMAT
+                + System.lineSeparator(), outContent.toString());
+    }
+
+    @Test
+    public void execute_longToValue_expectIllegalInputExceptionThrown() {
```
Comment (wongwh2002) [inline src/test/java/seedu/commands/FindCommandTest.java:119]: are we standardising to type of message thrown for the exception thrown?

Comment (TPH777) [reply src/test/java/seedu/commands/FindCommandTest.java:119]: Sure i can do it, I will standardise for all the other test case also then

**Comment by TPH777** [comment]
The changes made will overwrite alot of code, so author tag must be added before tuesday submission.

**Comment by NigelYeoTW** [comment]
wait out will cause a lot of merge conflict.

**Comment by TPH777** [comment]
> wait out will cause a lot of merge conflict.

Agree. Merge other PRs first. Its easier for me to settle others merge conflict.

**Comment by wongwh2002** [comment]
merged wx PR

**Comment by NigelYeoTW** [comment]
lgtm

**Pull request merged**